### PR TITLE
[release-v0.21.0] Update func version to v0.14.0

### PR DESCRIPTION
**Pull request by dsimansk**
The func version is point to current main. It should be updated once the release is published.

/hold
/cc @rhuss

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -14,7 +15,7 @@ require (
 	k8s.io/apimachinery v0.19.7
 	k8s.io/cli-runtime v0.19.7
 	k8s.io/client-go v0.19.7
-	k8s.io/code-generator v0.19.7
+	k8s.io/code-generator v0.20.1
```
Comment (dsimansk) [inline go.mod:18]: @rhuss I'm not quiet sure why `go mod tidy` bumps generator version as well. I've pinned it to match the rest of k8s dependencies.

Comment (rhuss) [reply go.mod:18]: yeah, that is always tricky to find out. I suspect that there is a direct reference to `code-generator` somewhere and the go dep resolution always picks the highest number.

Comment (rhuss) [reply go.mod:18]: However, since this is a build time dependency I don't expect any issue if the code compiles and test passes.

Comment (dsimansk) [reply go.mod:18]: Yep, we have the reference in hack/tools.go to help with Export type generation etc.

Comment (dsimansk) [reply go.mod:18]: So it's `buildpacks -> different version of go-containerregistry@v0.4.0 -> newer code-generator`. However, as you mentioned it's build time dependency and we don't rely on any code generation refresh here, should we keep the newer one?

Comment (dsimansk) [reply go.mod:18]: Per our chat keeping the knative/pkg version of code-generator https://github.com/openshift/knative-client/pull/694/commits/97666596f1186e8cf73b53f7908f6aad98dc4e56.

**Pull request merged**